numba/core/inline_closurecall.py

```diff
@@ -1068,10 +1068,10 @@ def is_removed(val, removed):
         # this doesn't work in objmode as it's effectively untyped
         if typed:
             len_func_var = ir.Var(scope, mk_unique_var("len_func"), loc)
-            from numba.cpython.rangeobj import range_iter_len
+            from numba.cpython.rangeobj import comprehension_iter_len
             stmts.append(_new_definition(func_ir, len_func_var,
-                        ir.Global('range_iter_len', range_iter_len, loc=loc),
-                        loc))
+                                         ir.Global('comprehension_iter_len', comprehension_iter_len, loc=loc),
```

Comment:
flake8---line too long
```suggestion
                                         ir.Global('comprehension_iter_len', 
                                                   comprehension_iter_len, 
                                                   loc=loc),
```

Reply:
Interesting. For me flake8 did not complain about this. But since we are still maybe changing the name I would wait. What do you think about the name?

Reply:
Some older files have been added to the exclude-list for flake8 because there are too much problems in it (which is unfortunate).

As for the name, I agree with https://github.com/numba/numba/issues/7394#issuecomment-916904842 and the function is not comprehension specific.

Reply:
I decided on the more correct `length_of_iterator`, since we don't input the `iterable`, but the actual `iterator` (the `iterable` would be the `range`/`list` itself)